Model/Source/CarrierInsurancePossibilities.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\Magento\Model\Source;
+
+use Exception;
+use Magento\Framework\Data\OptionSourceInterface;
+use MyParcelNL\Sdk\src\Factory\ConsignmentFactory;
+use MyParcelNL\Sdk\src\Model\Consignment\AbstractConsignment;
+use MyParcelNL\Sdk\src\Services\CountryCodes;
+
+class CarrierInsurancePossibilities implements OptionSourceInterface
+{
+    protected string $type;
+    protected AbstractConsignment $carrier;
+
+    /**
+     * @param string $carrierName
+     * @param string $type
+     * @throws Exception
+     */
+    public function __construct(string $carrierName, string $type)
+    {
+        $this->type = $type;
+        $this->carrier = consignmentFactory::createByCarrierName($carrierName);
```

Comment:
geeft je editor geen warning dat de casing niet klopt?
```suggestion
        $this->carrier = ConsignmentFactory::createByCarrierName($carrierName);
```

Reply:
Nee